components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1.common/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/common/mappings/ImportUtils.java

```diff
@@ -791,6 +816,96 @@ private static JsonElement retrieveValidatedDTOObject(String pathToArchive, Bool
         return configObject;
     }
 
+    /**
+     * Import Operation Policy as a zip.
+     *
+     * @param pathToArchive            Path to the extracted folder
+     * @param organization             Organization
+     * @param apiProvider              API Provider
+     * @throws APIManagementException If an error occurs while processing the policy files
+     */
+    public static OperationPolicyDataDTO importPolicy(String pathToArchive, String organization,
+            APIProvider apiProvider) throws APIManagementException {
+
+        OperationPolicySpecification policySpecification = null;
+
+        try {
+            OperationPolicyDefinition ccPolicyDefinition = null;
+            OperationPolicyDefinition synapseDefinition = null;
+
+            String[] fileLocations = pathToArchive.split("/");
+
+            // File names of all types should be the same
+            String fileName = File.separator + fileLocations[fileLocations.length - 1];
+            String policyDefinitionJsonContent = getPolicyFileContentAsJson(pathToArchive + fileName);
+            if (policyDefinitionJsonContent == null) {
+                throw new APIManagementException(
+                        "Cannot find Operation Policy definition. policyDefinition.yaml should present",
+                        ExceptionCodes.ERROR_FETCHING_DEFINITION_FILE);
+            }
+
+            policyDefinitionJsonContent = CommonUtil.yamlToJson(policyDefinitionJsonContent);
+            policySpecification = APIUtil.getValidatedOperationPolicySpecification(policyDefinitionJsonContent);
+
+            OperationPolicyData operationPolicyData = new OperationPolicyData();
+            operationPolicyData.setOrganization(organization);
+            operationPolicyData.setSpecification(policySpecification);
+
+            String pathToJ2File = pathToArchive + fileName + ImportExportConstants.SYNAPSE_EXTENSION;
+
+            String synapsePolicyDefinitionJsonContent = readDefinitionFiles(pathToJ2File);
+
+            if (synapsePolicyDefinitionJsonContent != null) {
+                synapseDefinition = new OperationPolicyDefinition();
```

Comment:
Shall we refactor this code segment to a small function (for a sample one, refer below) and use that function here and 871 line?

public OperationPolicyDefinition setPropertiesToOperationPolicyDefinition(String policyDefinitionJsonContent, String type, OperationPolicyData operationPolicyData) {
                OperationPolicyDefinition operationPolicyDefinition = new OperationPolicyDefinition();
                operationPolicyDefinition.setContent(policyDefinitionJsonContent);
                operationPolicyDefinition.setGatewayType(type);
                operationPolicyDefinition.setMd5Hash(APIUtil.getMd5OfOperationPolicyDefinition(operationPolicyDefinition));
                operationPolicyData.setSynapsePolicyDefinition(operationPolicyDefinition);
}

Reply:
Fixed by [b7f664a](https://github.com/wso2/carbon-apimgt/pull/11325/commits/b7f664af4e393bb1fe5d307dd9dc9c9107e372d6)